### PR TITLE
Fix loadWithXhr

### PR DESCRIPTION
**Pull request by hpinkos**
Addresses this comment: https://github.com/AnalyticalGraphicsInc/cesium/pull/6023#discussion_r156082866

@mramato

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hpinkos, thanks for the pull request! Maintainers, we have a signed CLA from @hpinkos, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Core/loadWithXhr.js [hunk 1]
```diff
@@ -187,13 +187,13 @@ define([
...
-            //All modern browsers will go into either the first if block or last else block.
-            //Other code paths support older browsers that either do not support the supplied responseType
-            //or do not support the xhr.response property.
-            if (defined(response) && (!defined(responseType) || (browserResponseType === responseType))) {
-                deferred.resolve(response);
-            } else if (xhr.status === 204) { // accept no content
+            if (xhr.status === 204) { // accept no content
                 deferred.resolve();
+            } else if (defined(response) && (!defined(responseType) || (browserResponseType === responseType))) {
+                //All modern browsers will go into either the first if block or last else block.
```
Comment (mramato) [inline Source/Core/loadWithXhr.js]: This comment is still wrong.  Move it back to the top of the if and specify that `modern browsers will go into either the first two if blocks`

**Comment by hpinkos** [comment]
@mramato ready

**Comment by mramato** [comment]
Thanks.

**Pull request merged**